packages/usb-adapter/src/index.ts

```diff
@@ -54,26 +55,42 @@ export default class USBAdapter extends Adapter<[]> {
     });
   }
 
+  static isPrinter(device: usb.Device): boolean {
+    try {
+      const length = device.configDescriptor?.interfaces.filter((iface) => {
+        return iface.filter((conf) => {
+          return conf.bInterfaceClass === IFACE_CLASS.PRINTER;
+        }).length;
+      }).length;
+      return (length !== undefined && length > 0);
+    }
+    catch (e) {
+      // console.warn(e)
+      return false;
+    }
+  }
+
   static findPrinter() {
-    return getDeviceList().filter((device) => {
+    return getDeviceList().filter((device) => USBAdapter.isPrinter(device));
+  }
+
+  static getDevice(vid: number, pid: number) {
+    return new Promise((resolve, reject) => {
       try {
-        return device.configDescriptor?.interfaces.filter((iface) => {
-          return iface.filter((conf) => {
-            return conf.bInterfaceClass === IFACE_CLASS.PRINTER;
-          }).length;
-        }).length;
+        const device = findByIds(vid, pid);
+        device?.open();
+        resolve(device);
       }
-      catch (e) {
-        // console.warn(e)
-        return false;
+      catch (err) {
+        reject(err);
       }
     });
   }
 
-  static getDevice(vid: number, pid: number) {
-    return new Promise((resolve, reject) => {
+  static getDeviceBySerial(serialNumber: string) {
```

Comment:
There is a breaking change. Can we use another approach that adapts the current API? Otherwise, we need to bump a major version for these changes. ( I prefer the former. :P )

Reply:
Hmm i don't know what Github is showing here. The getDevice function still exists in line 76 and was not touched during my changes. If i view the changes locally only the findPrinter function was adapted to
``` Typescript
static findPrinter() {
    return getDeviceList().filter((device) => USBAdapter.isPrinter(device));
  }
```
And then i have added getDeviceBySerial without touching the getDevice() function. If you look at the whole file you will see it :)